src/nodes.coffee

```diff
@@ -854,6 +854,10 @@ exports.Block = class Block extends Base
       body, directives
     }
 
+  astLocationData: ->
+    return if @isRootBlock and not @locationData?
```

Comment:
When there are no source tokens, the root `Block` ends up not having any `locationData` assigned to it (which was crashing `jisonLocationDataToAstLocationData()`)

So here we just avoid calling the root `astLocationData()` method for this case

And then the AST ends up with the correct location data on the `File` and `Program` AST nodes because of the [existing logic](https://github.com/jashkenas/coffeescript/blob/ast/src/coffeescript.coffee#L121) that sets the AST location data for those root nodes after-the-fact